sky/resources.py

```diff
@@ -966,20 +966,22 @@ def _try_validate_labels(self) -> None:
         """
         if not self._labels:
             return
-
-        if self.cloud is None:
-            # Because each cloud has its own label format, we cannot validate
-            # the labels without knowing the cloud.
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError(
-                    'Cloud must be specified when labels are provided.')
-
-        # Check if the label key value pairs are valid.
+        if self.cloud is not None:
+            validated_clouds = [self.cloud]
+        else:
+            # If no specific cloud is set, validate label against ALL clouds.
+            # The label will be dropped if invalid for any one of the cloud
+            validated_clouds = clouds.CLOUD_REGISTRY.values()
```

Comment:
It might be better to only verify for the clouds that are enabled by the user: `enabled_clouds = sky_check.get_cached_enabled_clouds_or_refresh()`, so people are not limited by the clouds that they are not using.